cartography/intel/crxcavator/crxcavator.py

```diff
@@ -0,0 +1,220 @@
+def get_users_extensions(crxcavator_api_key, crxcavator_base_url):
+    """
+    Gets listing of all users who have installed each extension
+    :param crxcavator_api_key: The API key to access the CRXcavator service
+    :param crxcavator_base_url: The URL for the CRXcavator API
+    :return: JSON text blob containing user email to extension id mapping defined at
+    https://crxcavator.io/apidocs#tag/group/paths/~1group~1users~1extensions/get
+    """
+    return call_crxcavator_api("/group/users/extensions", crxcavator_api_key, crxcavator_base_url)
+
+
+def call_crxcavator_api(api_and_parameters, crxcavator_api_key, crxcavator_base_url):
+    """
+    Perform the call requested to the CRXcavator API
+    :param crxcavator_api_key: The API key to access the CRXcavator service
+    :param crxcavator_base_url: The URL for the CRXcavator API
+    :param api_and_parameters: Query string for the API including the required parameters
+    :return: Returns JSON text blob for the API called. API spec is at https://crxcavator.io/apidocs
+    """
+    uri = crxcavator_base_url + api_and_parameters
+    data = requests.get(
+        uri,
+        headers={'Accept': 'application/json',
+                 'API-Key': crxcavator_api_key})
+    # if call failed, use requests library to raise an exception
+    data.raise_for_status()
+    return data.json()
+
+
+def transform_extensions(extension_json):
+    """
+    Transforms the raw extensions JSON from the API into a list of extensions data
+    :param extension_json:  The JSON text blob returned from the CRXcavator API
+    :return: List containing extension info for ingestion
+    """
+    # the JSON returned from the CRXcavator API does not return well formatted objects
+    # instead, each object is named after it's key, making enumeration more difficult
+    # will build a cleaner object for import into graph
+
+    _data_index = 1
+    _row_index = 0
+
+    extensions = []
+    for extension in extension_json.items():
+        details = extension[_data_index][_row_index]
+        if not details:
+            continue
+        extension_id = details['extension_id']
+        version = details['version']
+        data = details['data']
+        # permissions warnings can be a list or null, if list concat to a single string
+        permission_warnings = data['webstore'].get('permission_warnings')
+        if permission_warnings:
+            permission_warnings = ', '.join(permission_warnings)
+        extensions.append({
+            'id': "{0}|{1}".format(extension_id, version),
+            'extension_id': extension_id,
+            'version': version,
+            'risk_total': data['risk'].get('total'),
+            'risk_metadata': json.dumps(data['risk'].get('metadata')),
+            'address': data['webstore'].get('address'),
+            'email': data['webstore'].get('email'),
+            'icon': data['webstore'].get('icon'),
+            'last_updated': data['webstore'].get('last_updated'),
```

Comment:
[schema-block] let's avoid issues with our internal lastupdated and rename to crxcavator_last_update for example